### ISSUE TITLE
WorldGuard region lookup & claim prevention

### ISSUE THREAD

**Issue by RHX0R3** [WasabiThumb/xclaim#110]
Dear author and contributors,

I want to request a compatibility feature, that implements checks for a **[WorldGuard](https://builds.enginehub.org/job/worldguard)** region.
_In mind, that **WorldGuard** still is the most used protection plugin out there._

The best option could be an implementation of a `flag` for regions, to deny claims being made while inside or partially overlapping.

This flag could be active (`default: true`) so it allows user claims. And can be manually set to `false` in case, you don't want to let players claim chunks, for example at spawn.

### Example flag:

```
xclaim-allowed
```

Also, an optional permission handling could be, that you can't claim a chunk, even if you have all permissions (e. g. `*`).
Alternatively, this could be controlled with a option inside the configuration file with something like (in case, someone wants to override them):

### Example option:

```
options:
  worldguard:
    permissions-override-flag: false
    # default: false
```

As last time;

I know, that this is not a top priority, as it is just a enhancement / feature request.
Also, you have any right to dismiss that idea.
But I'd like to see that in a future update, and I believe, that other users would like that too.

Thank you for reading!

_Best regards,
X0R3_

**Comment by WasabiThumb** [comment]
The current behavior is that claims are not allowed if it intersects _any_ WorldGuard region in any way. This was added in [1.10.1](https://github.com/WasabiThumb/xclaim/releases/tag/1.10.1) and made default in [1.20.2](https://github.com/WasabiThumb/xclaim/releases/tag/1.10.2) (so if you started using the plugin within one of these transitional versions, you may have to update your config).

Could you explain why it may be worth implementing an "allows" flag on regions? When would we want an XClaim claim within a WorldGuard region?

**Comment by RHX0R3** [comment]
Hello Wasabi,

well. Then I have to update my config file.
I did not test it yet, because I did not see the option in the file yet.

Of course, that would be enough - a seperate flag would not be as useful, when all regions are blocked already.

The only case I could imagine, a flag would be useful, is when you have a region with high priority inside a region with low priority.
Like AreaShop.

Nonetheless, feel free to close this issue - as it is not important or needed.

Sorry again~
